pkg/compute/bidder.go

```diff
@@ -47,110 +53,160 @@ func NewBidder(params BidderParams) Bidder {
 	}
 }
 
-// TODO: evaluate the need for async bidding and marking bids as waiting
-func (b Bidder) RunBidding(ctx context.Context, request AskForBidRequest, usageCalc capacity.UsageCalculator) {
-	var (
-		// ask the bidding strategy if we should bid on this job
-		bidStrategyRequest = bidstrategy.BidStrategyRequest{
-			NodeID:   b.nodeID,
-			Job:      *request.Execution.Job,
-			Callback: b.getApproveURL(),
-		}
-
-		routingMetadata = RoutingMetadata{
-			// the source of this response is the bidders nodeID.
+func (b Bidder) ReturnBidResult(
+	ctx context.Context,
+	localExecutionState store.LocalExecutionState,
+	response *bidstrategy.BidStrategyResponse,
+) {
+	if response.ShouldWait {
+		return
+	}
+	result := BidResult{
+		RoutingMetadata: RoutingMetadata{
 			SourcePeerID: b.nodeID,
-			// the target of this response is the source of the request.
-			TargetPeerID: request.SourcePeerID,
-		}
-		executionMetadata = NewExecutionMetadata(request.Execution)
-	)
+			TargetPeerID: localExecutionState.RequesterNodeID,
+		},
+		ExecutionMetadata: NewExecutionMetadata(localExecutionState.Execution),
+		Accepted:          response.ShouldBid,
+		Reason:            response.Reason,
+	}
+	b.callback.OnBidComplete(ctx, result)
+}
 
-	response, resourceUsage, err := b.doBidding(ctx, bidStrategyRequest, usageCalc)
+type BidderRequest struct {
+	SourcePeerID string
+	// Execution specifies the job to be executed.
+	Execution *models.Execution
+	// WaitForApproval specifies whether the compute node should wait for the requester to approve the bid.
+	// if set to true, the compute node will not start the execution until the requester approves the bid.
+	// If set to false, the compute node will automatically start the execution after bidding and when resources are available.
+	WaitForApproval bool
 
-	// update the execution with the calculated resource usage
-	request.Execution.AllocateResources(request.Execution.Job.Task().Name, resourceUsage)
+	// ResourceUsage specifies the requested resources for this execution
+	ResourceUsage *models.Resources
+}
 
+// TODO: evaluate the need for async bidding and marking bids as waiting
```

Comment:
This should be a ticket rather than a TODO

Reply:
https://github.com/bacalhau-project/bacalhau/issues/3732